source/arm7/libc/crt0_device_list.c

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Zlib
+//
+// Copyright (C) 2025 Edoardo Lolletti (edo9300)
+
+#include <stddef.h>
+#include <string.h>
+
+#include <nds/system.h>
+#include <nds/device_list.h>
+
+static size_t getDeviceNameLenFromAppName(const char (appname)[0x40])
+{
+	for(int i = 0; i < 0x40; ++i)
+	{
+		if(appname[i] == ':')
+		{
+			return i;
+		}
+	}
+	return 0;
+}
+
+struct device_list_argv {
+	struct __argv base;
+    char argv[0x42];
+};
+
+void check_device_list()
+{
+	if (!isDSiMode())
+		return;
+
+	if (__system_argv->argvMagic == ARGV_MAGIC)
```

Comment:
This will take up a lot of space in DS-only programs, not just DSi. Move the code of `check_device_list()` to something like `check_device_list_internal()` and do this:

```
TWL_CODE void check_device_list_internal(void)
{
    ...
}

void check_device_list(void)
{
	if (!isDSiMode())
		return;

    check_device_list_internal();
}
```